client/src/app/categories-page/categories-page.component.ts

```diff
@@ -0,0 +1,145 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+
+import { Category } from '../global-components/models/category.model';
+
+import { CategoriesService } from './services/categories.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'app-categories-page',
+  templateUrl: './categories-page.component.html',
+  styleUrls: ['./categories-page.component.scss']
+})
+
+export class CategoriesPageComponent implements OnInit {
+
+  verifyTitle: string = '';
+  verifyMessage: string = '';
+  deletedCategoryId!: number;
+  categories: Category[] = [];
+  toasterMessage: string = '';
+  verifyModalStatus: boolean = false;
+  isAddCategoryError: boolean = false;
+  addCategoryErrorMessage: string = '';
+  toasterMessageStatus: boolean = false;
+  isOpenAddCategoryModal: boolean = false;
+  selectedCategoryType: string = 'expense';
+  isActiveIncomeCategories: boolean = false;
+  isActiveExpenseCategories: boolean = false;
+  isWarning: boolean = false;
+
+  addCategoryForm = this.fb.group({
+    categoryType: ['expense'],
+    title: ['', [Validators.required]]
+  })
+
+  constructor(
+    private fb: FormBuilder,
+    private categoryService: CategoriesService
+  ) { }
+
+  ngOnInit(): void {
+    this.getCategories();
+  }
+
+  getCategories(type: any = null, searchedText: string = '') {
+    this.categoryService
+      .getCategories(type)
+      .pipe(untilDestroyed(this))
+      .subscribe({
+        next: data => {
+          this.categories = data.categories
+          this.categories = this.categories.filter(category => category.title.includes(searchedText))
+        }
+      })
+  }
+
+  addCategory() {
+    if (this.addCategoryForm.valid) {
+      const { categoryType, title } = this.addCategoryForm.value;
+      this.categoryService
+        .addCategory(categoryType, title)
+        .pipe(untilDestroyed(this))
+        .subscribe({
+          next: data => {
+            this.toasterMessageStatus = true;
+            this.toasterMessage = data.message;
+            this.isOpenAddCategoryModal = false;
+            setTimeout(() => {
+              this.toasterMessageStatus = false;
+              window.location.reload()
```

Comment:
You shouldn't reload the page.

Reply:
When I use route.navigateByUrl() it was not working So I used window.location.reload() but I will try to solve this bug

Reply:
Pls solve